File: frontend/src/types/CorrectionResult.ts
import type { TrackRow } from "./TrackRow";

/** What one backwards correction did, and the track as it now stands. */
export interface CorrectionResult {
  readonly from_ordinal: number;
  readonly to_ordinal: number;
  /** How far the position dropped — what the toast reports. */
  readonly removed_units: number;
  /** Rows deleted outright. A row trimmed rather than deleted is not counted. */
  readonly removed_advances: number;
  /** False when the destination was already the position and nothing was written. */
  readonly moved: boolean;
  readonly track: TrackRow;
}

File: backend/src/sidra/api/models/correction_result.py
from __future__ import annotations

from pydantic import BaseModel

from sidra.api.models.track_row import TrackRow


class CorrectionResult(BaseModel):
    """What one backwards correction did, and the track as it now stands."""

    from_ordinal: int
    to_ordinal: int

    removed_units: int
    """How far the position dropped, which is what the toast reports."""

    removed_advances: int
    """Rows deleted outright. A row trimmed rather than deleted is not counted."""

    moved: bool
    """False when the destination was already the position and nothing was written."""

    track: TrackRow
